pallets/xcm-asset-manager/src/lib.rs

```diff
@@ -0,0 +1,404 @@
+// Copyright 2020-2022 Litentry Technologies GmbH.
+// This file is part of Litentry.
+//
+// Litentry is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Litentry is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Litentry.  If not, see <https://www.gnu.org/licenses/>.
+
+//! Inspired by:
+//! - Moonbeam `pallet_asset_manager`
+//! implementations.
+
+// (1) The local asset support is removed. This pallet is for managing Xcm foreign asset
+// specifically. (2) The mapping of AssetId and AssetType is now capable of manual choosen
+// (3) Compatibility to orml_token instead of pallet_assets
+// (4) Code for destroy asset is removed
+
+//! TODO Doc comments for the pallet
+//! # Asset Manager Pallet
+//!
+//! This pallet allows to register new assets if certain conditions are met
+//! The main goal of this pallet is to allow Litmus to register XCM assets
+//! and control the creation of foreign assets
+//! The assumption is we work with AssetTypes, which can then be comverted to AssetIds
+//!
+//! This pallet has five storage items:
+//! - AssetIdType: A mapping from AssetId->AssetType.
+//! - AssetIdUnitsPerSecond: An AssetId->u128 mapping that holds how much each AssetId should be
+//!   charged per unit of second, in the case such an Asset is received as a XCM asset.
+//! - AssetTypeId: A mapping from AssetType -> AssetId.
+//! - ForeignAssetTracker: The counter of foreign assets that have been created so far.
+//! - AssetIdMetadata: An AssetId->AssetMetadata mapping that holds the metadata token info.
+//!
+//! This pallet has six extrinsics:
+//! - register_foreign_asset: Register a foreign asset in this pallet.
+//! - set_asset_units_per_second: Set the unit per second that should be charged for a particular
+//!   asset.
+//! - relocate_foreign_asset_id: Allow to manipulate ForeignAssetTracker value.
+//! - update_foreign_asset_metadata: Update the metadata of asset.
+//! - add_asset_type: Add the correspondence between AssetId and AssetType.
+//! - remove_asset_type: Remove the correspondence between AssetId and AssetType. At least one
+//!   relation must exist.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+#![allow(clippy::clone_on_copy)]
+
+// TODO implement
```

Comment:
This comment can be removed(?) or do we still need something?

Reply:
removed.